### DISCUSSION TITLE
[bug]pkg打包后，运行环境里的文件夹检查结果不准确。

### DISCUSSION THREAD

**Discussion by pkjy** [midwayjs/midway#4166] [General / 闲聊]
https://github.com/midwayjs/midway/blob/696137ffef10971f519bb838fb9e904a3f5bc58d/packages/static-file/src/middleware/static.middleware.ts#L62

这行代码检查了静态资源文件夹是否存在，使用了
https://github.com/midwayjs/midway/blob/696137ffef10971f519bb838fb9e904a3f5bc58d/packages/core/src/util/fs.ts#L5
`fs.access`来判断。
pkg打包后会生成[`/snapshot`](https://github.com/vercel/pkg?tab=readme-ov-file#snapshot-filesystem)的一个虚拟目录。
实测发现该方法检查不到pkg运行时的虚拟文件夹。
我把上面的62行的判断注释后，能正常访问到静态资源。否则会一直报错 
`DirectoryNotFoundError: Path /snapshot/project/public not exist, please check it.`
建议改为fs.readdirSync来判断，我这个case里试了下是可行的。

以下是从gpt那获得的一些建议。

![QQ_1730772599215](https://github.com/user-attachments/assets/560fbfbc-631b-4d46-9bc1-aece412b29c5)

**Comment by pkjy** [comment]
https://github.com/midwayjs/midway/pull/4167

**Comment by czy88840616** [comment]
最好是能判断出是 pkg 环境，只在 static-file 中做一些处理，收敛影响范围。

**Reply by pkjy** [reply]
判断pkg环境倒是没啥问题，因为现在用的工具是封装在core里的，就看怎么去调整了。

`const isPkg = typeof process.pkg !== 'undefined';`

**Reply by czy88840616** [reply]
https://github.com/midwayjs/midway/pull/4180
